homework/unique_ptr/unique_ptr.hpp

```diff
@@ -0,0 +1,111 @@
+#pragma once
+
+#include <cstddef>
+
+template <typename T>
+class unique_ptr {
+public:
+    unique_ptr();
+    unique_ptr(std::nullptr_t);
+    explicit unique_ptr(T*);
+    unique_ptr(unique_ptr const&);
```

Comment:
```suggestion
    unique_ptr(unique_ptr const&) = delete;
```
We don't want to have a copy constructor

Reply:
In the task description I read "Constructor copies a pointer" so I thought I'd need a copy constructor.

Reply:
Removed anyway.

Reply:
After watching 'Zarządzanie zasobami #2' I understood that unique must be unique (no copying allowed).